crates/bifrost/src/providers/replicated_loglet/loglet.rs

```diff
@@ -260,6 +260,15 @@ impl<T: TransportConnect> Loglet for ReplicatedLoglet<T> {
         )
         .run(self.networking.clone())
         .await?;
+        // If we are the sequencer, we need to wait until the sequencer is drained.
+        if let SequencerAccess::Local { handle } = &self.sequencer {
+            handle.drain().await?;
+            self.known_global_tail.notify_seal();
```

Comment:
Why is it important to drain before sending the seal notification?

Reply:
What we don't want is:
- Append goes in flight
- Append is successful, known_global_tail is not updated yet.
- Seal is done on log-servers
- We choose current known_global_tail as sealed
- Append is acknowledged, known_global_tail is updated.

This leads to a find_tail that conveys the picture as if no appends will be acknowledged after this offset, that's the critical guarantee that we must maintain.